core/i18n/request.ts

```diff
@@ -11,7 +13,12 @@ export default getRequestConfig(async ({ requestLocale }) => {
   }
 
   return {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-    messages: (await import(`../messages/${locale}.json`)).default,
+    locale,
+    messages: deepmerge(
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-unsafe-member-access
+      (await import(`../messages/${fallbackLocale}.json`)).default,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-unsafe-member-access
+      (await import(`../messages/${locale}.json`)).default,
+    ),
```

Comment:
Can we skip this if `locale === fallbackLocale`?

Reply:
Yep, good point :) ... I was so focused on having a fallback different from local that I missed the most obvious scenario ;). I will fix it right now.